src/domain/validation/FlowContainerValidator.ts

```diff
@@ -182,63 +196,81 @@ function blockTypeToInterfaceName(type: string): string | null {
 }
 
 /**
- * Check that all resources asked for within blocks are available in the Resources array of the container
+ * Check that all resources asked for within blocks are available in the Resources array
  * @param container Flow package container
  * @returns null if all resources are available, otherwise an array of the missing resource UUIDs
  */
 function checkAllResourcesPresent(container: IContainer): string[] | null {
   const resourcesRequested: string[] = []
+  let allResources: IResources = []
+  if (container.specification_version < '1.0.0-rc4') {
+    // for these version, resources is no longer accessible in IContainer, but we still want to check them if requested
+    allResources = get(container, 'resources') as IResources
+  } else {
+    allResources = []
+  }
+
   container.flows.forEach(flow => {
+    if (container.specification_version >= '1.0.0-rc4') {
+      allResources.push(...flow.resources)
+    }
+
     flow.blocks.forEach(block => {
-      if (block.type == 'MobilePrimitives.Message') {
```

Comment:
Extracted into `collectResourceUuidsFromBlock()`

Reply:
I wonder if there is a better way to do this by indicating in the interfaces when something should be a resource uuid (rather than just having `string`). Probably a bit complicated but this looks fragile to spec changes. Maybe a longer term goal/something worth spiking?

Reply:
@jcbashdown , I'm afraid I don't really understand this, could you be more specific pls ?
Are you suggesting we turn this https://github.com/FLOIP/flow-runner/blob/master/src/model/block/ISelectOneResponseBlockConfig.ts#L23 into `IResource['uuid']` instead of `string` for eg. ?

Reply:
What I'm basically asking for is a single source of truth for what is a resource uuid that this function can then use instead of using these if statements. It would be easy to miss this function if the spec changes and then things will break in a non obvious way. I don't have a good solution in mind myself, would using `Resource['uuid']` help do you think?